code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
+  if (testTime === undefined) { testTime = Date.now() }
+  // // calculate counter from testTime
+  const counter = Math.floor(testTime / INTERVAL)
+  const otp = ONEUtil.genOTP({ seed: wallet.seed, counter })
+  // // calculate wallets effectiveTime (creation time) from t0
+  const info = await wallet.wallet.getInfo()
+  const t0 = new BN(info[3]).toNumber()
+  const walletEffectiveTime = t0 * INTERVAL
+  const index = ONEUtil.timeToIndex({ effectiveTime: walletEffectiveTime, time: testTime })
+  const eotp = await ONE.computeEOTP({ otp, hseed: wallet.hseed })
+
+  // Format commit and revealParams based on tokenType
+  let commitParams
+  let revealParams
+  let paramsHash
+  console.log(`operationType: ${operationType}`)
+  switch (operationType) {
+    case ONEConstants.OperationType.TRACK:
+    case ONEConstants.OperationType.UNTRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, dest }
+      revealParams = { operationType, tokenType, contractAddress, dest }
+      break
+    case ONEConstants.OperationType.OVERRIDE_TRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      revealParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      break
+    case ONEConstants.OperationType.RECOVER_SELECTED_TOKENS:
```

Comment:
dest and data

Reply:
Done 🙏 
However this test is still failing. It may have been I just missed something in the test but here's where I got to debugging.
Debugged  it appears to be failing in `Reveal.sol`-> `authenticate`  which calls `authenticateCores`
Wondering whether `RECOVER_SELECTED_TOKENS` should call `isCorrectRecoveryProof` as well
```
        uint32 coreIndex = 0;
        if (op.operationType == Enums.OperationType.RECOVER) {
            coreIndex = isCorrectRecoveryProof(core, oldCores, auth);
        } else {
            coreIndex = isCorrectProof(core, oldCores, auth);
 ```
 Either that or I just missed something in `general.ts` with how we test `RECOVER_SELECTED_TOKENS`

```
Error message
       OPERATION 9 RECOVER_SELECTED_TOKENS: must be able to recover selected tokens:
     Error: Returned error: VM Exception while processing transaction: revert Param mismatch
      at commitReveal (test/util.js:415:44)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at Object.executeStandardTransaction (test/util.js:376:3)
      at Context.<anonymous> (test/general.js:629:5)
```